### PR TITLE
Sizing yaml template for SystemLink Enterprise pilots

### PR DESCRIPTION
**Pull request by BKnight760**
Provide a sizing file that specifies resources and replicas for a SystemLink Enterprise pilot deployment.

- [x] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/install-systemlink-enterprise/blob/master/CONTRIBUTING.md).

### What does this Pull Request accomplish?

This pull request adds a new template yaml file that specifies the resources and replica counts for a pilot-sized deployment.

### Why should this Pull Request be merged?

Provide customers with a template to speed up initial deployment of pilot environments.

### What testing has been done?

Deployed the solution manually and verified that the services start correctly and are capable of handling a reasonable load for a pilot.

### CODE REVIEW

File: getting-started/templates/pilot-sizing.yaml [hunk 1]
```diff
@@ -0,0 +1,891 @@
...
+      maxReplicas: 2
+      targetCPUUtilizationPercentage: 60
+      targetMemoryUtilizationPercentage: 70
+
+argoworkflows:
+  argo-workflows:
+    server:
+      replicas: 1
+
+feedservice:
```
Comment (JD-Robertson) [inline getting-started/templates/pilot-sizing.yaml:809]: suggestion: Alphabetize services to make this file  easier to navigate?

File: getting-started/templates/pilot-sizing.yaml [hunk 2]
```diff
@@ -0,0 +1,891 @@
+# Default values for SystemLink pilot resources and replicas.
```
Comment (JD-Robertson) [inline getting-started/templates/pilot-sizing.yaml:1]: Do we want some documentation as to what this is and why someone would use it?

Do we want internal-facing documentation on how we arrived at these numbers?

File: getting-started/templates/pilot-sizing.yaml [hunk 3]
```diff
@@ -0,0 +1,891 @@
...
+    requests:
+      cpu: "100m"
+      memory: "256Mi"
+    limits:
+      memory: "512Mi"
+
+minio:
+  persistence:
+    size: 1Gi
+  provisioning:
```
Comment (JD-Robertson) [inline getting-started/templates/pilot-sizing.yaml:16]: Provisioning only works if you enable minio clustering. (`mode: distributed`). That's not the default for the top-level chart or the values file in this folder. This configuration will be ignored in favor of the defaultBuckets setting.

File: getting-started/templates/pilot-sizing.yaml (hunk 1 above)
Comment (BKnight760) [reply getting-started/templates/pilot-sizing.yaml]: I pushed a new commit with the services alphabetized.

File: getting-started/templates/pilot-sizing.yaml (hunk 2 above)
Comment (BKnight760) [reply getting-started/templates/pilot-sizing.yaml]: I updated the comments at the top of the file to better describe what this file does and how to use it.

**Review by prestwick — CHANGES_REQUESTED** [review]
I know my comment about Mongo might risk our end date, but there is interest in deprecating our support for cluster internal Mongo for SLE. Given that this is likely to happen we should strong consider not using that configuration for pilots.

File: getting-started/templates/pilot-sizing.yaml (hunk 2 above)
Comment (BKnight760) [reply getting-started/templates/pilot-sizing.yaml]: Discussed with Mark.  I will upload the Jupyter notebooks I used for generating the data in order to test the UI experience at this small scale.

File: getting-started/templates/pilot-sizing.yaml [hunk 4]
```diff
@@ -0,0 +1,895 @@
+# Value overrides for SystemLink pilot resources and replicas.
+#
+# Use this file in addition to the `systemlink-values.yaml` to
+# customize the SystemLink deployment for small pilot clusters.
```
[the remark this answers is not in the document]
Comment (BKnight760) [reply getting-started/templates/pilot-sizing.yaml:4]: Done.

File: getting-started/templates/pilot-sizing.yaml [hunk 5]
```diff
@@ -0,0 +1,895 @@
...
+#
+# This file reduces the number of service replicas and pod CPU and
+# memory requests such that the application can deploy with fewer
+# kubernetes cluster nodes. The cluster is intended to be deployed
+# with the following node topology:
+# 4 nodes - 8 CPU, 32 GB memory - General node pool
+# 2 nodes - 8 CPU, 32 GB memory - Dedicated for Dremio
+
+# Two replicas are used for most services for increased performance.
+# The Mongo databases are reduced to a single replica and as a result,
```
[the remark this answers is not in the document]
Comment (BKnight760) [reply getting-started/templates/pilot-sizing.yaml]: Added a comment about pilots using managed MongoDB Atlas or MongoDB Enterprise Advanced.

File: getting-started/templates/pilot-sizing.yaml (hunk 3 above)
Comment (BKnight760) [reply getting-started/templates/pilot-sizing.yaml]: As per our offline discussion, I will leave this configuration.

**Review by JD-Robertson — APPROVED** [review]

**Review by prestwick — APPROVED** [review]

**Pull request merged**